File: examples/authentication/cc/rt/naeem/hottentot/runtime/service/default_request_callback.cc
#include <iostream>

#include "default_request_callback.h"
#include "request_handler.h"

#include "../response.h"
#include "../request.h"
#include "../logger.h"


namespace naeem {
  namespace hottentot {
    namespace runtime {
      namespace service {
        Response*
        DefaultRequestCallback::OnRequest(void *source,
                                          Request &request) {
          ::naeem::hottentot::runtime::Logger::GetOut() << "A new request is received." << std::endl;
          if (requestHandlers_->count(request.GetServiceId()) > 0) {
            Response *response = new Response;
            RequestHandler *requestHandler = requestHandlers_->find(request.GetServiceId())->second;
            if (requestHandler == 0) {
              return 0;
            }
            requestHandler->HandleRequest(request, *response);
            return response;
          } else {
            return 0;            
          }
        }
      }
    }
  }
}

File: examples/authentication/java/src/main/java/runtime/DefaultRequestCallback.java
package runtime;




import java.util.Map;

public class DefaultRequestCallback implements RequestCallBack {

    private Map<Integer , RequestHandler> requestHandlers;
    public DefaultRequestCallback(Map<Integer,RequestHandler> requestHandlers) {
        this.requestHandlers = requestHandlers;
    }

    public Response onRequest(Request request){
        //TODO(ali)
        RequestHandler requestHandler = requestHandlers.get((int)request.getServiceId());
        System.out.println(request);
        return requestHandler.handleRequest(request);
    }
}
